### ISSUE TITLE
Google Drive is storing the file in the root, ignoring the paths

### ISSUE THREAD

**Issue by kelter-antunes** [SharpGrip/FileSystem#42]
Hi,
It seems that GDrive adapter is not following the paths like azure blob and Amazon s3 are doing it. For those two everything is working as expected but not for GDrive.

```
// Adapter settings
string adapterPrefix = "gdrive";
string rootPath = "/demofolder";

// File system path settings
string path = "/from/demo/app";
string fileName = "example.env";
string localFilePath = @"C:\Users\XXXXXX\Downloads\example.env";

...

// Create Google Drive adapter
var googleDriveAdapter = new GoogleDriveAdapter(adapterPrefix, rootPath, driveService);

// Create file system with Google Drive adapter
googleDriveFileSystem = new FileSystem(new List<IAdapter> { googleDriveAdapter });

...

// Construct virtual path
virtualPath = $"{adapterPrefix}://{path}{fileName}"; // =  "gdrive://from/demo/app/example.env"

// Write file to Google Drive
googleDriveFileSystem.WriteFile(virtualPath, binaryData, true);

// Retrieve file from Google Drive
var existingFile = googleDriveFileSystem.GetFile(virtualPath);

```

The file is stored in the GDrive root:
![image](https://github.com/user-attachments/assets/cc686b7a-3756-4cb3-824d-d3632923b5d4)

And then when trying to get the file for the same virtualPath I get the following exception:

**"File 'demofolder/from/demo/app/example.env' not found in adapter with prefix 'gdrive'."**

Is it me that I'm doing anything wrong?

**Comment by mvdgun** [comment]
@kelter-antunes could you check what would happen if you add a forward slash between the path and filename?

`virtualPath = $"{adapterPrefix}://{path}/{fileName}";`

**Comment by kelter-antunes** [comment]
![image](https://github.com/user-attachments/assets/ad3a823f-4cc5-48c6-a3bd-b941b1790094)

Makes it double slash, and then there's an exception when trying to save the file:
GoogleApiException: The service drive has thrown an exception. HttpStatusCode is InternalServerError. Internal Error

**Comment by mvdgun** [comment]
@kelter-antunes, after some investigation, it appears that the Google Drive adapter does not implicitly handle creating folders in a path if they do not exist. A current workaround is to manually create the necessary folders before storing files. I will leave this ticket open to address implicit folder creation in future versions.

**Comment by kelter-antunes** [comment]
Yes, I can confirm that. I tested it myself, and if the **full path** doesn't exist, it places the file in the root. I tried it with just one or two folders from the path, and if the full path doesn't exist, it always places the file in the root.
Thank you for your quick response.

**Comment by kelter-antunes** [comment]
@mvdgun  I just wanted to share some more thoughts on this. I'm also looking to store files on Shared Drives, and it's getting complicated since we can't have a full path for a Shared Drive (at least to my knowledge). In the end, Google Drive, for better or worse, only requires a Folder ID to place the file, whether it's on our local drive or a Shared Drive.

I can't think of the best solution to maintain the same generic actions across all adapters. However, knowing this, it would be nice if, for the Google Drive adapter, we could pass only a Folder ID instead of a root + virtual path or something similar.

**Comment by mvdgun** [comment]
@kelter-antunes The Google Drive adapter has a lot of custom directory traversing to emulate a file system like behavior (loke you said with the parent folder ID). I would suspect that Shared drives would work the same as a regular Google Drive though. I don't have a lot of free time on my hands currently so I cannot quickly test this one out.
